api/types/role.go

```diff
@@ -479,14 +479,48 @@ func (r *RoleV6) convertDenyKubernetesResourcesBetweenRoleVersions(resources []K
 		return resources
 	default:
 		v7resources := slices.Clone(resources)
+		var extraResources []KubernetesResource
 		for i, r := range v7resources {
+			// "namespace" kind used to mean "namespaces" and all resources in the namespace.
+			// It is now represented by 'namespaces' for the resource itself and wildcard for
+			// all resources in the namespace.
+			if r.Kind == KindKubeNamespace {
+				r.Kind = Wildcard
+				r.Namespace = r.Name
+				r.Name = Wildcard
+				r.APIGroup = Wildcard
+				v7resources[i] = r
+				extraResources = append(extraResources, KubernetesResource{
+					Kind:  "namespaces",
+					Name:  r.Namespace,
+					Verbs: r.Verbs,
+				})
+				continue
+			}
+			// The namespace field was ignored in v7 for global resources.
+			if r.Namespace != "" && slices.Contains(V7KubernetesClusterWideResourceKinds, r.Kind) {
```

Comment:
Why do we need this check here?

Reply:
Because in rolev7, the namespace field was ignored for global resources, i.e. when having `kind: '*', namespace: 'foo'`, rolev7 would still match clusterroles & co. To maintain the behavior, we clear the namespace value when dealing with global resources.